scripts/mbedtls_framework/test_driver.py

```diff
@@ -0,0 +1,219 @@
+        "Clone partially builtin tree, rewrite header inclusions and prefix"
+        "exposed C identifiers.")
+
+    parser.add_argument("dst_dir", metavar="DST_DIR",
+                        help="Destination directory.\n"
+                        " - If absolute, used as-is.\n"
+                        " - If relative, interpreted relative to the repository root.\n")
+    parser.add_argument("--driver", default="libtestdriver1", metavar="DRIVER",
+                        help="Test driver name (default: %(default)s).")
+    return parser
+
+def iter_code_files(root: Path) -> Iterable[Path]:
+    """
+    Iterate over all "*.c" and "*.h" files found recursively under the `include`
+    and `src` subdirectories of `root`.
+    """
+    for directory in ("include", "src"):
+        directory_path = root / directory
+        for ext in (".c", ".h"):
+            yield from directory_path.rglob(f"*{ext}")
+
+def run_ctags(file: Path) -> Set[str]:
+    """
+    Extract the C identifiers in `file` using ctags.
+
+    Identifiers of the following types are returned (with their corresponding
+    ctags c-kinds flag in parentheses):
+
+    - macro definitions (d)
+    - enum values (e)
+    - functions (f)
+    - enum tags (g)
+    - function prototypes (p)
+    - struct tags (s)
+    - typedefs (t)
+    - union tags (u)
+    - global variables (v)
+    """
+
+    result = subprocess.run(
+        ["ctags", "-x", "--language-force=C", "--c-kinds=defgpstuv", str(file)],
+        capture_output=True, text=True, check=True, encoding="utf-8"
+    )
+    identifiers = set()
+    for line in result.stdout.splitlines():
+        identifiers.add(line.split()[0])
+
+    return identifiers
+
+class TestDriverGenerator:
+    """A TF-PSA-Crypto test driver generator"""
+    def __init__(self, dst_dir: Path, driver: str):
+        self.dst_dir = dst_dir
+        self.driver = driver
+        # Path of 'dst_dir'/include/'driver'
+        self.test_driver_include_dir = None #type: Path | None
+
+    def build_tree(self, src_dir: Path, exclude_files: Optional[Set[str]] = None) -> None:
+        """
+        Build a test driver tree from `src_dir`.
+
+        The source directory `src_dir` is expected to have the following structure:
+        - an `include` directory
+        - an `src` directory
+        - the `include` directory contains exactly one subdirectory
+
+        Only the `include` and `src` directories from `src_dir` are used to build
+        the test driver tree, and their directory structure is preserved.
+
+        Only "*.h" and "*.c" files are copied. Files whose names match any of the
+        patterns in `exclude_files` are excluded.
+
+        The subdirectory inside `include` is renamed to `driver` in the test driver
+        tree, and header inclusions are adjusted accordingly.
+        """
+        include = src_dir / "include"
+        if not include.is_dir():
+            raise RuntimeError(f'Do not find "include" directory in {src_dir}')
```

Comment:
Minor: “do not find” is an instruction. A complaint should be “Did not find” or more usually “… not found”.